topiary-config/src/language.rs

```diff
@@ -235,117 +236,24 @@ impl GitSource {
-                None
-            }
         };
 
-        let mut config = cc::Build::new();
-        config.cpp(true).opt_level(3).cargo_metadata(false);
-        config.target(BUILD_TARGET);
-        config.host(BUILD_TARGET);
-
-        let compiler = config.get_compiler();
-        let mut command = Command::new(compiler.path());
-        command.current_dir(src_path);
-        for (key, value) in compiler.env() {
-            command.env(key, value);
-        }
-
-        command.args(compiler.args());
-
-        command
-            .arg("-shared")
-            .arg("-fPIC")
-            .arg("-fno-exceptions")
-            .arg("-I")
-            .arg(header_path)
-            .arg("-o")
-            .arg(&target_path);
-
-        if let Some(scanner_path) = scanner_path.as_ref() {
-            if scanner_path.extension() == Some("c".as_ref()) {
-                command.arg("-xc").arg("-std=c11").arg(scanner_path);
-            } else {
-                let mut cpp_command = Command::new(compiler.path());
-                cpp_command.current_dir(src_path);
-                for (key, value) in compiler.env() {
-                    cpp_command.env(key, value);
-                }
-                cpp_command.args(compiler.args());
-                let object_file = target_path.with_file_name(format!("{}_scanner.o", &self.rev));
-                cpp_command
-                    .arg("-fPIC")
-                    .arg("-fno-exceptions")
-                    .arg("-I")
-                    .arg(header_path)
-                    .arg("-o")
-                    .arg(&object_file)
-                    .arg("-std=c++14")
-                    .arg("-c")
-                    .arg(scanner_path);
-                let output = cpp_command.output()?;
-                if !output.status.success() {
-                    return Err(TopiaryConfigFetchingError::Subprocess(format!(
-                        "{:#?}, {:#?}",
-                        output.stdout, output.stderr
-                    )));
-                }
-
-                command.arg(&object_file);
-            }
-        }
-
-        command.arg("-xc").arg("-std=c11").arg(parser_path);
-
-        if cfg!(all(
-            unix,
-            not(any(target_os = "macos", target_os = "illumos"))
-        )) {
-            command.arg("-Wl,-z,relro,-z,now");
-        }
-
-        let output = command.output()?;
-
-        if !output.status.success() {
-            return Err(TopiaryConfigFetchingError::Subprocess(format!(
-                "{:#?}, {:#?}",
-                String::from_utf8_lossy(&output.stdout),
-                String::from_utf8_lossy(&output.stderr),
-            )));
-        }
-
-        log::info!("{}: successfully compiled", name);
+        // Build grammar
+        log::info!("{name}: Building grammar");
+        let mut loader =
+            tree_sitter_loader::Loader::new().map_err(TopiaryConfigFetchingError::Build)?;
+        loader.debug_build(false);
```

Comment:
Any reason to do a debug build  here?

Reply:
AIUI, `Loader::debug_build` doesn't do anything besides set a flag. So `debug_build(false)` means "don't do a debug build".

Reply:
Ah, I see. A peculiar choice of API :upside_down_face: